src/series/errorBase.js

```diff
@@ -0,0 +1,99 @@
+import d3 from 'd3';
+import fractionalBarWidth from '../util/fractionalBarWidth';
+
+export default function() {
+
+    var xScale = d3.time.scale(),
+        yScale = d3.scale.linear(),
+        yHigh = d3.functor(0),
+        xHigh = function(d, i) { return d.date; },
+        yLow = d3.functor(0),
+        xLow = function(d, i) { return d.date; },
+        xValue = function(d, i) { return d.date; },
+        yValue = function(d, i) { return d.y; },
+        barWidth = fractionalBarWidth(0.75),
+        xValueScaled = function(d, i) {
+            return xScale(xValue(d, i));
+        };
+
+    function base() { }
+
+    base.width = function(data) {
+        return barWidth(data.map(xValueScaled));
+    };
+
+    base.values = function(d, i) {
+        return {
+            x: xValueScaled(d, i),
+            y: yScale(yValue(d, i)),
+            yLow: yScale(yLow(d, i)),
+            yHigh: yScale(yHigh(d, i)),
+            xLow: xScale(xLow(d, i)),
+            xHigh: xScale(xHigh(d, i))
+        };
+    };
+    base.defined = function(d, i) {
+        return yLow(d, i) != null && yHigh(d, i) != null &&
+            xValue(d, i) != null && yValue(d, i) != null
+            && xLow(d, i) != null && xHigh(d, i) != null;
+    };
+
```

Comment:
I'm pretty relaxed about line breaks, as long as it is readable and doesn't fail a linting rule, I'm happy!